cni/network/network_test.go

```diff
@@ -1246,6 +1247,21 @@ func (n *InterfaceGetterMock) GetNetworkInterfaces() ([]net.Interface, error) {
 	return n.interfaces, nil
 }
 
+func (n *InterfaceGetterMock) GetNetworkInterfaceAddrs(iface *net.Interface) ([]net.Addr, error) {
```

Comment:
We should be able to pass into the InterfaceGetterMock struct the potential []net.Addrs to return from this function and based on the interface, return a particular []net.Addr based on some field in the net.Interface struct. What you have here gives us a mocked response, but what if we have more complex tests that require us to return certain addresses for one particular interface, and another set of addresses for a different interface?

Reply:
seems like we can only take advantage of index from net.Interface...